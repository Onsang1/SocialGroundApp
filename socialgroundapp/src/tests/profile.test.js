/**
* @jest-environment jsdom
*/
import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
// import testing library functions
// import userEvent from '@testing-library/user-event';
import { expect, test } from '@jest/globals';
import renderer from 'react-test-renderer';
// import { within } from '@testing-library/dom';
import '@testing-library/jest-dom/extend-expect';
import Profile from '../components/profile';

test('renders Profile link', () => {
  render(
    <Router>
      <Profile />
    </Router>,
  );
  const linkElement = screen.getByText(/Posts/);
  expect(linkElement).toBeInTheDocument();
});

// snapshot testing

//TODO: update snapshot
test('SocialGround profile page matches snapshot', () => {
  const component = renderer.create(
    <Router>
      <Profile />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
