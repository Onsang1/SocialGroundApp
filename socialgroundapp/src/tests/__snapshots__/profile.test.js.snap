// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialGround profile page matches snapshot 1`] = `
<div
  id="react-root"
>
  <div
    aria-disabled="false"
    role="button"
  >
    <a
      href="/feed"
      onClick={[Function]}
    >
      <img
        alt="sg-logo-small"
        className="sg-logo-small"
        src={Object {}}
      />
    </a>
  </div>
  <div
    className="main-body"
  >
    <div
      className="profile-bar"
    >
      <div
        aria-disabled="false"
        role="button"
      >
        <img
          alt="default-user"
          className="default-user"
          placeholder={Object {}}
          src={Object {}}
        />
      </div>
      <span
        className="profile-username"
      />
      <span
        className="posts"
      >
        Posts
      </span>
      <span
        className="followers"
      >
        Followers
      </span>
      <span
        className="following"
      >
        Following
      </span>
      <span
        className="posts-count"
      />
      <a
        href="/Find"
        onClick={[Function]}
      >
        <span
          className="followers-count"
        />
      </a>
      <a
        href="/Find"
        onClick={[Function]}
      >
        <span
          className="following-count"
        />
      </a>
      <div
        aria-disabled="false"
        role="button"
      >
        <a
          href="/upload"
          onClick={[Function]}
        >
          <img
            alt="circled-plus"
            className="circled-plus"
            src={Object {}}
          />
        </a>
      </div>
      <div
        aria-disabled="false"
        role="button"
      >
        <a
          href="/setting"
          onClick={[Function]}
        >
          <img
            alt="user-setting"
            className="user-setting"
            src={Object {}}
          />
        </a>
      </div>
    </div>
    <div
      className="boxes"
    >
      <a
        href="/post/undefined"
        onClick={[Function]}
      >
        <div
          className="box1"
        >
          <img
            alt=""
            height="100%"
            width="100%"
          />
        </div>
      </a>
      <a
        href="/post/undefined"
        onClick={[Function]}
      >
        <div
          className="box2"
        >
          <img
            alt=""
            height="100%"
            width="100%"
          />
        </div>
      </a>
      <a
        href="/post/undefined"
        onClick={[Function]}
      >
        <div
          className="box3"
        >
          <img
            alt=""
            height="100%"
            width="100%"
          />
        </div>
      </a>
      <a
        href="/post/undefined"
        onClick={[Function]}
      >
        <div
          className="box4"
        >
          <img
            alt=""
            height="100%"
            width="100%"
          />
        </div>
      </a>
      <a
        href="/post/undefined"
        onClick={[Function]}
      >
        <div
          className="box5"
        >
          <img
            alt=""
            height="100%"
            width="100%"
          />
        </div>
      </a>
      <a
        href="/post/undefined"
        onClick={[Function]}
      >
        <div
          className="box6"
        >
          <img
            alt=""
            height="100%"
            width="100%"
          />
        </div>
      </a>
      <a
        href="/post/undefined"
        onClick={[Function]}
      >
        <div
          className="box7"
        >
          <img
            alt=""
            height="100%"
            width="100%"
          />
        </div>
      </a>
      <a
        href="/post/undefined"
        onClick={[Function]}
      >
        <div
          className="box8"
        >
          <img
            alt=""
            height="100%"
            width="100%"
          />
        </div>
      </a>
      <a
        href="/post/undefined"
        onClick={[Function]}
      >
        <div
          className="box9"
        >
          <img
            alt=""
            height="100%"
            width="100%"
          />
        </div>
      </a>
    </div>
  </div>
</div>
`;
